### PR TITLE
Chef Automate HA Pre-Requisites

### PR DESCRIPTION
**Pull request by dishanktiwari2501**
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/docs-chef-io/content/automate/chef-automate-ha-prerequisites.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+The Automate HA supports three types of deployment:
+
+* On-Premise Deployment
+* AWS Deployment Deployment
+* AWS Managed Services Deployment
+
+The below requirements are elaborated according to the above three deployments.
+
+## On-Premise Deployment Pre-Requisite
```
Comment (IanMadd) [inline components/docs-chef-io/content/automate/chef-automate-ha-prerequisites.md:25]: It's "on-premises", not "on-premise". There's no such thing as "on-premise"

File: components/docs-chef-io/content/automate/chef-automate-ha-prerequisites.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+{{< warning >}}
+{{% automate/ha-warn %}}
+{{< /warning >}}
+
+{{< warning >}}
+The below pre-requisites are according to our organizational standard. If you are using any specified version not mentioned here or a third party extensions or software you can reach out to the CAMs and our Customer Support Team.
+{{< /warning >}}
+
+Before installing Chef automate HA on On-premise deployment, ensure you have taken a quick tour of this pre-requisite page.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/chef-automate-ha-prerequisites.md:23]: `Before installing Chef automate HA on On-premise deployment` can you rewrite this sentence plz

File: components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+{{< warning >}}
+{{% automate/ha-warn %}}
+{{< /warning >}}
+
+{{< warning >}}
+The below pre-requisites are according to our organizational standard. If you are using any specified version not mentioned here or a third party extensions or software you can reach out to the CAMs and our Customer Support Team.
+{{< /warning >}}
+
+Before installing Chef automate HA on AWS deployment, ensure you have taken a quick tour of this pre-requisite page.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md:23]: `Before installing Chef automate HA on AWS deployment` - can you rewrite this sentence?

File: components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+{{< note >}}
+
+- For **OpenSearch** and **PostgresSQL**, a minimum of three node clusters is required.
+- For production, OpenSearch volume size also depends on the number of nodes and frequency of Chef Infra Client runs and compliance scans.
+{{< /note >}}
+
+### Load Balancer
+
+LoadBalancers in AWS deployment are set up according to [Chef Automate HA Architecture](/automate/ha/).
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md:63]: LoadBalancers will be provisioned automatically, customer does not need to create manually in this kind of deployment

File: components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    echo "aws_secret_access_key=<SECRET_KEY>" >> ~/.aws/credentials
+    echo "region=<AWS-REGION>" >> ~/.aws/credentials
+    ```
+
+- Have DNS certificate ready in ACM for 2 DNS entries: Example: `chefautomate.example.com`, `chefinfraserver.example.com`
+    Reference for [Creating new DNS Certificate in ACM](/automate/ha_aws_cert_mngr/)
+- Have SSH Key Pair ready in AWS, so new VM's are created using that pair.
+    Reference for [AWS SSH Key Pair creation](https://docs.aws.amazon.com/ground-station/latest/ug/create-ec2-ssh-key-pair.html)
+- We do not support passphrase for Private Key authentication.
+- Preferred key type will be ed25519=
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md:110]: please remove the `=` at the end

File: components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+- DNS is configured to redirect `chefautomate.example.com` to the Primary Load Balancer.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md:111]: let's discuss these.

File: components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+{{< warning >}}
+{{% automate/ha-warn %}}
+{{< /warning >}}
+
+{{< warning >}}
+The below pre-requisites are according to our organizational standard. If you are using any specified version not mentioned here or a third party extensions or software you can reach out to the CAMs and our Customer Support Team.
+{{< /warning >}}
+
+Before installing Chef automate HA on AWS managed services deployment, ensure you have taken a quick tour of this pre-requisite page.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md:23]: please rewrite this stmt

File: components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+{{< note >}}
+
+- For **OpenSearch** and **PostgresSQL**, a minimum of three node clusters is required.
+- For production, OpenSearch volume size also depends on the number of nodes and frequency of Chef Infra Client runs and compliance scans.
+{{< /note >}}
+
+### Load Balancer
+
+LoadBalancers in aws managed services deployment are set up according to [Chef Automate HA Architecture](/automate/ha/).
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md:63]: LoadBalancers will be provisioned automatically in this mode of deployment

File: components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+## Deployment Specific Pre-requisites
+
+The AWS Managed Services deployment specific pre-requisites are as follows:
+
+- Virtual Private Cloud (VPC) should be created with internet gateway attached in AWS before starting. Reference for [VPC and CIDR creation](/automate/ha_vpc_setup/)
+- If you want to use Default VPC, then you have to create Public and Private Subnet, if subnet are not available. Please refer [this](https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html)
+- We need 3 private and 3 public subnet in a vpc (1 subnet for each AZ). As of now we support dedicate subnet for each AZ.
+- We recommend to create a new VPC. And Bastion should be in the same VPC.
+- Setup AWS RDS Postgresql 13.5 in the same VPC where we have the basion and automate ha node going to be created. Click [here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_GettingStarted.CreatingConnecting.PostgreSQL.html) to know more.
+- Please refer to the [table](/automate/chef-automate-ha-prerequisites/#migration/) to identify the correct OpenSearch version to be installed.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md:91]: Now this stmt is not needed

File: components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+- Setup AWS OpenSearch with the identified version in the same VPC where we have the basion and automate ha node going to be created. Click [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/createupdatedomains.html) to know more.
```
Comment (Dmaddu) [inline components/docs-chef-io/content/automate/ha-aws-managed-services-deployment-prerequisites.md:92]: Mention the AWS OpenSearch version as 1.3

File: components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+
+**Ports for Bastion before deployment**
+| Port No. | Outgoing to | Incoming from |
+|----------|-------------|---------------|
+| 22       | Subnet      | All           |
+| 80       |             | Internet      |
+| 443      |             | Internet      |
+
+
+**Ports mapping after deployment:**
```
Comment (vivekshankar1) [inline components/docs-chef-io/content/automate/ha-aws-deployment-prerequisites.md:132]:
```suggestion
**Port mappings required before deployment:**
```

**Review by IanMadd — CHANGES_REQUESTED** [review]
- All instances of "on-premise" should be replaced with "on-premises". That includes file names.
- Use underscores in filenames for consistency.
- There are three deployment methods, so there should be three deployment prerequisite pages.
- Change the menu titles to "Prerequisites"
- The value for the `gh_repo` property is just `automate`.
- You've got a few instances of "aws". AWS is an acronym so it's always capitalized unless you're talking about the CLI tool. 
- The menu parent and identifier should match the identifiers here: https://github.com/chef/chef-web-docs/pull/4053/files
- The menus for the three deployment pages should also be updated to use the same parent identifiers:
   - ha_onprim_deployment_procedure
   - ha_aws_deploy_steps
   - ha_aws_managed_deploy_steps

**Comment by kalroy** [comment]
@dishanktiwari2501  Please create a story for all the feedbacks from @IanMadd . Merging it for the urgent release.

**Pull request merged**